Plots/Polygons/NCL_polyg_19.py

```diff
@@ -0,0 +1,200 @@
+
+        if pdata >= colorbounds[len(colorbounds)-1]:
+            return colormap.colors[x-1]
+        if pdata >= colorbounds[x]:
+            continue
+        else:
+            # Index is 'x-1' because colorbounds is one item longer than colormap
+            return colormap.colors[x-1]
+
+###############################################################################
+# Define helper function to remove ticks from axes
+
+
+def removeTicks(axis):
+
+    axis.get_xaxis().set_visible(False)
+    axis.get_yaxis().set_visible(False)
+
+###############################################################################
+# Define helper function to plot and color each state
+
+
+def plotRegion(shape, axis, xlim):
+
+    # Plot each shape within a region (ex. mainland Alaska and all of it's surrounding Alaskan islands)
+    for i in range(len(shape.shape.parts)):
+
+        i_start = shape.shape.parts[i]
+        if i == len(shape.shape.parts)-1:
+            i_end = len(shape.shape.points)
+        else:
+            i_end = shape.shape.parts[i+1]
+
+        x = []
+        y = []
+        patches = []
+
+        for i in shape.shape.points[i_start:i_end]:
+            if xlim[0] is not None and i[0] < xlim[0]:
+                continue
+            if xlim[1] is not None and i[0] > xlim[1]:
+                continue
+            else:
+                x.append(i[0])
+                y.append(i[1])
+
+        # Fill each state with color
+        abbrevname = shape.record.STUSPS
+        pop = nameandpopdict[abbrevname]
+        color = findDivColor(colorbounds, pop)
+
+        axis.plot(x, y, color='black', linewidth=0.3)
+        patches.append(Polygon(np.vstack((x, y)).T, True, linewidth=0.3, color=color))
+
+        pc = PatchCollection(patches, match_original=True, edgecolor='k', linewidths=1., zorder=2)
+        axis.add_collection(pc)
+
+
+###############################################################################
+# Plot:
+
+# Create figure
+fig = plt.figure(figsize=(8, 8))
+spec = gridspec.GridSpec(ncols=1, nrows=2, hspace=0.05, wspace=0.1, figure=fig, height_ratios=[2, 1])
+
+# Create upper axis
+ax1 = fig.add_subplot(spec[0, 0], frameon=False)
+removeTicks(ax1)
+
+# Create lower axis
+ax2 = fig.add_subplot(spec[1, 0], frameon=False)
+removeTicks(ax2)
+
+# Create three inset axes on lower axis for Alaska, Hawaii, and Puerto Rico
+axin1 = ax2.inset_axes([0.0, 0.7, 0.30, 0.80], frameon=False)
+removeTicks(axin1)
+axin2 = ax2.inset_axes([0.40, 0.7, 0.20, 0.40], frameon=False)
+removeTicks(axin2)
+axin3 = ax2.inset_axes([0.70, 0.7, 0.30, 0.30], frameon=False)
+removeTicks(axin3)
+
+# Specify which shapes not to include when plotting the mainland U.S.
+nonmainlandus = ['Commonwealth of the Northern Mariana Islands', 'Guam', 'Puerto Rico', 'Hawaii', 'Alaska', 'American Samoa',
+                 'United States Virgin Islands']
+
+# Get population of each state
+nameandpopdict = getStatePopulations(statepopulationfile)
```

Comment:
You've used this variable name in a few places, and it's hard to read. Again I would separate the words using underscores